numpy/core/_internal.py

```diff
@@ -16,7 +15,7 @@
 except ImportError:
     ctypes = None
 
-IS_PYPY = platform.python_implementation() == 'PyPy'
+IS_PYPY = 'PyPy' in sys.version
```

Comment:
[PEP 421](https://peps.python.org/pep-0421/#required-attributes) added `sys.implementation` for python3 
```suggestion
IS_PYPY = sys.implementation.name == 'pypy'
```

Reply:
@mattip Thanks! I didn't know about that one. Do you want me to also change the implementation of`numpy.testing.IS_PYPY` with `sys.implementation.name == 'pypy'`?
(and then remove the test as well?)

Reply:
> Do you want me to also change the implementation ...

Makes sense, sure.